app/user.go

```diff
@@ -1105,7 +1105,9 @@ func (a *App) UpdateUser(user *model.User, sendNotifications bool) (*model.User,
 			return nil, model.NewAppError("UpdateUser", "store.sql_user.update.email_taken.app_error", nil, "user_id="+user.Id, http.StatusBadRequest)
 		}
 
-		user.Email = prev.Email
+		if !user.IsBot {
```

Comment:
Can we leave a comment here as to why this line exists? This whole function is a bit confusing and this just adds to it.

Reply:
New comment added. Let me know if that seems reasonable. And I agree this was very confusing

Reply:
"To allow email verification..." is that really what we area allowing here? I seems like we are allowing bot account email changes not email verification.

Reply:
That's true.  I'll reword and re-commit.